tests/test_models.py

```diff
@@ -76,49 +86,81 @@ def test_search(as_ops):
         json_object = [{'uri': '1234'}]
         url = f'/repositories/{repo_id}/search?'
         m.get(url, json=json_object)
-        results = as_ops.search(string, repo_id, rec_type, note_type)
+        results = as_ops.search(string, repo_id, rec_type, logger, note_type)
         for result in results:
             assert result == json_object[0]['uri']
 
 
-def test_save_record(as_ops, caplog):
+def test_save_record(as_ops, logger, caplog):
     """Test post_record method."""
+    caplog.set_level(logging.INFO)
     with requests_mock.Mocker() as m:
         rec_obj = models.Record()
         uri = '/repositories/2/resources/423'
         dry_run = 'False'
         rec_obj['uri'] = uri
         json_object = {'post': 'Success'}
         m.post(uri, json=json_object)
-        as_ops.save_record(rec_obj, dry_run)
+        as_ops.save_record(rec_obj, dry_run, logger)
         message = json.loads(caplog.messages[0])['event']
         assert message == json_object
 
 
-def test_save_record_flushes_changes(as_ops):
+def test_save_record_flushes_changes(as_ops, logger):
     with requests_mock.Mocker() as m:
         uri = '/foo/bar/1'
         dry_run = 'False'
         m.post(uri, json={'post': 'Success'})
         r = models.Record({'uri': uri})
         r['title'] = 'A title'
         assert r.modified
-        as_ops.save_record(r, dry_run)
+        as_ops.save_record(r, dry_run, logger)
         assert not r.modified
 
 
-def test_get_aos_for_resource(as_ops):
+def test_get_aos_for_resource(as_ops, logger):
     """Test get_aos_for_resource function."""
     with requests_mock.Mocker() as m:
         resource = '/repositories/2/resources/423'
         json_object = {'record_uri': '/archival_objects/1234', 'children':
                        [{'record_uri': '/archival_objects/5678'}]}
         m.get(f'{resource}/tree', json=json_object)
-        aolist = as_ops.get_aos_for_resource(resource)
+        aolist = as_ops.get_aos_for_resource(resource, logger)
         assert '/archival_objects/5678' in aolist
 
 
-def test_extract_fields(as_ops):
+def test_audit():
+    """"Test audit function."""
+    sample_record = {'uri': '123'}
+    change = {'op': 'add', 'path': '/title', 'value': 'I am a title'}
+    msg = models.audit(record=sample_record, **change)
+    assert msg == {'uri': '123', 'field': '/title', 'old': None,
+                   'new': 'I am a title'}
+
+
+def test_concat_id():
+    """"Test concat_id function."""
+    rec_obj = {'id_0': '1', 'id_1': '2', 'id_2': '3', 'id_3': '4'}
+    coll_id = models.concat_id(rec_obj)
+    assert coll_id == '1-2-3-4'
+
+
+# How to test this?
+# def test_create_csv_from_log():
+#   """"Test create_csv_from_log function."""
+#     assert False
+
+
+def test_download_json():
+    """Test download_json function."""
+    rec_obj = models.Record()
+    rec_obj['uri'] = '/test/123'
+    path = models.download_json(rec_obj)
+    assert os.path.isfile(path)
+    os.remove(path)
```

Comment:
There's a problem here in that if the assertion fails for any reason, the `os.remove` call won't be made. There's no easy solution here as it would require rethinking how several pieces work. I would say it's also a relatively unlikely scenario and it's not a huge problem if it were to happen. Ideally, we'd like to avoid this kind of thing, but for now, I'm just going to mark this as a thing we may want to revisit at some point.